src/common/utils.js

```diff
@@ -140,6 +140,74 @@ function promiseAny(promises: Promise<any>[]): Promise<any> {
   })
 }
 
+/**
+ * Waits for the promises to resolve and uses a provided checkResult function
+ * to return a key to identify the result. The returned promise resolves when
+ * n number of promises resolve to identical keys.
+ */
+async function promiseNy<T, I>(
+  promises: Promise<T>[],
+  checkResult: T => I | void,
+  n?: number = promises.length
+): Promise<T> {
```

Comment:
Remove `I` type variable in favor of `checkResult: T => string | void`; object keys can only ever be strings, so the variable here is not needed.